fqd_dss.c

```diff
@@ -63,7 +63,11 @@ fqd_worker_thread(void *arg)
 
   while (!worker_thread_shutdown) {
     if (!CK_FIFO_SPSC_ISEMPTY(q)) {
-      if (ck_fifo_spsc_dequeue(q, &m)) {
+      bool success;
+      ck_fifo_spsc_dequeue_lock(q);
```

Comment:
I am not certain we need dequeue locks on the read side as each worker thread has it's own queue.   But they can't hurt.

The write side definitely needs locks as we write into these per thread queues.